src/deploy/functions/tasks.ts

```diff
@@ -0,0 +1,279 @@
+      });
+      const pollerOptions: OperationPollerOptions = Object.assign(
+        {
+          pollerName: `update-${fn.name}`,
+          operationResourceName: updateRes.name,
+          onPoll,
+        },
+        defaultPollerOptions
+      );
+      const operationResult = await pollOperation<CloudFunctionTrigger>(pollerOptions);
+      params.timer.endTimer(fn.name);
+      helper.printSuccess(fn.name, "update");
+      return operationResult;
+    } catch (err) {
+      params.errorHandler.record("error", fn.name, "update", err.message || "");
+    }
+  };
+}
+
+export function deleteFunctionTask(params: TaskParams, fnName: string) {
+  return async () => {
+    utils.logBullet(
+      clc.bold.cyan("functions: ") +
+        "deleting function " +
+        clc.bold(helper.getFunctionLabel(fnName)) +
+        "..."
+    );
+    params.timer.startTimer(fnName, "delete");
+    try {
+      const deleteRes = await cloudfunctions.deleteFunction({
+        functionName: fnName,
+      });
+      const pollerOptions: OperationPollerOptions = Object.assign(
+        {
+          pollerName: `delete-${fnName}`,
+          operationResourceName: deleteRes.name,
+        },
+        defaultPollerOptions
+      );
+      const operationResult = await pollOperation<void>(pollerOptions);
+      params.timer.endTimer(fnName);
+      helper.printSuccess(fnName, "delete");
+      return operationResult;
+    } catch (err) {
+      params.errorHandler.record("error", fnName, "delete", err.message || "");
+    }
+  };
+}
+
+export function upsertScheduleTask(
+  params: TaskParams,
+  fn: CloudFunctionTrigger,
+  appEngineLocation: string
+): () => Promise<any> {
+  return async () => {
+    const job = helper.toJob(fn, appEngineLocation, params.projectId);
+    try {
+      await cloudscheduler.createOrReplaceJob(job);
+      helper.printSuccess(fn.name, "upsert schedule");
+    } catch (err) {
+      params.errorHandler.record("error", fn.name, "upsert schedule", err.message || "");
+    }
+  };
+}
+
+export function deleteScheduleTask(
+  params: TaskParams,
+  fnName: string,
+  appEngineLocation: string
+): () => Promise<void> {
+  return async () => {
+    const jobName = helper.getScheduleName(fnName, appEngineLocation);
+    const topicName = helper.getTopicName(fnName);
+    try {
+      await cloudscheduler.deleteJob(jobName);
+    } catch (err) {
+      // If the job has already been deleted, don't throw an error.
+      if (err.status !== 404) {
+        params.errorHandler.record("error", fnName, "delete schedule", err.message || "");
+      }
+      logger.debug(`Scheduler job ${jobName} not found.`);
+    }
+    try {
+      await deleteTopic(topicName);
+      helper.printSuccess(fnName, "delete schedule");
+    } catch (err) {
+      // If the topic has already been deleted, don't throw an error.
+      if (err.status !== 404) {
+        params.errorHandler.record("error", fnName, "delete schedule", err.message || "");
```

Comment:
do we need to add `return;` here? Otherwise, the debug message below will always be printed.